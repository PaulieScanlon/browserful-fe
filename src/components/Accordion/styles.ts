import styled from 'react-emotion';
import { common, colours, spaceMd } from '../../theme';

interface IProps {
  maxHeight: string;
}

export const AccordionWrapper = styled.div({
  label: 'acorrdion-wrapper',
  position: 'relative',
  width: '100%',
  boxShadow: common.materialBoxShadow1
});

export const Input = styled.input<IProps>(
  {
    label: 'input',
    position: 'absolute',
    top: '-10px',
    left: '-10px',
    display: 'none',

    ':checked + label': {
      '&:after': {
        content: `""`,
        width: '14px',
        height: '14px',
        borderWidth: '4px',
        borderColor: `${colours.pink}`,
        backgroundColor: `${colours.white}`,
        color: colours.pink
      }
    }
  },
  ({ maxHeight }) => ({
    ':checked ~ div': {
      maxHeight: maxHeight
    }
  })
);

export const Label = styled.label({
  label: 'label',
  position: 'relative',
  display: 'block',
  backgroundColor: 'white',
  cursor: 'pointer',
  padding: `${spaceMd}px`,
  color: colours.greyDark,
  borderBottom: `1px solid ${colours.greyLight}`,
  transition: `${common.transition}`,
  boxSizing: 'border-box',
  ':after': {
    content: `""`,
    position: 'absolute',
    width: '20px',
    height: '20px',
    borderWidth: '1px',
    borderStyle: 'solid',
    borderRadius: '100%',
    right: `${spaceMd}px`,
    textAlign: 'center',
    borderColor: `${colours.greyLight}`,
    transition: `${common.transition}`,
    color: colours.greyLight
  }
});

export const ContentWrapper = styled.div({
  label: 'content-wrapper',
  overflow: 'hidden',
  maxHeight: '0px',
  transition: `${common.transition}`,
  padding: `0px ${spaceMd}px`,
  backgroundColor: colours.offWhite,
  color: colours.greyDark
});

export const Content = styled.div({
  label: 'content',
  padding: `${common.paddingTB} 0px`
});
