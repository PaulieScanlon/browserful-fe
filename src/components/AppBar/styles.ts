import styled from 'react-emotion';

import { colours, common } from '../../theme';

export const AppBarWrapper = styled.header({
  label: 'app-bar-wrapper',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
  backgroundColor: colours.white,
  boxShadow: common.materialBoxShadow1,
  minHeight: common.appBarHeight
});
