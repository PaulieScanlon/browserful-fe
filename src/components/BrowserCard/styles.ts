import styled from 'react-emotion';
import { common, colours, spaceMd } from '../../theme';

interface IProps {
  isExpanded?: boolean;
}

export const BrowserCardWrapper = styled.div({
  label: 'browser-card-wrapper',
  backgroundColor: colours.white,
  boxShadow: common.materialBoxShadow1
});

export const BrowserCardHeader = styled.div<IProps>(
  {
    label: 'browser-card-header',
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    margin: `0 ${spaceMd}px`,
    padding: `${spaceMd}px 0px`,
    borderBottomWidth: '1px',
    borderBottomStyle: 'solid'
  },
  ({ isExpanded }) => ({
    borderBottomColor: isExpanded ? colours.greyLight : colours.white
  })
);

export const BrowserCardHeaderSection = styled.div({
  label: 'browser-card-header-section',
  display: 'inline-flex',
  alignItems: 'center'
});

export const BrowserCardBody = styled.div<IProps>(
  {
    label: 'browser-card-body',
    overflow: 'hidden'
  },
  ({ isExpanded }) => ({
    height: isExpanded ? '100%' : '0px',
    padding: isExpanded ? `${spaceMd}px ${spaceMd}px` : `0px ${spaceMd}px`
  })
);
