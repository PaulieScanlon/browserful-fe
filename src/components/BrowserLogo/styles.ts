import styled from 'react-emotion';

export const BrowserLogoImage = styled.img({
  label: 'browser-logo-image'
});
