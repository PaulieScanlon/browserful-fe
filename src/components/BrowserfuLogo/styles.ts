import styled from 'react-emotion';

export const LogoWrapper = styled.div({
  label: 'logo-wrapper'
});
