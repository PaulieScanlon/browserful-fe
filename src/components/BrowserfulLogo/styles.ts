import styled from 'react-emotion';

import { common, spaceLg } from '../../theme';

export const LogoWrapper = styled.div({
  label: 'logo-wrapper',
  display: 'inline-flex',
  alignItems: 'center',
  marginLeft: `${spaceLg}px`
});

export const SvgWrapper = styled.span({
  label: 'svg-wrapper',
  display: 'inline-flex',
  marginRight: '5px'
});
