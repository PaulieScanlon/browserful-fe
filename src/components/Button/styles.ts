import styled from 'react-emotion';

import { scaffolding } from '../../theme';
import { font } from '../../typography';

interface IProps {
  fontColour: string;
  backgroundColour: string;
  grow: boolean;
}

export const ButtonElement = styled.button<IProps>(
  {
    label: 'button-element',
    display: 'inline-flex',
    justifyContent: 'center',
    padding: `${scaffolding.gutterLg} ${scaffolding.gutterXl}`,
    fontSize: font.fontSize,
    fontFamily: font.fontFamily,
    border: 0,
    borderRadius: '5px',
    outline: 'none',
    cursor: 'pointer'
  },
  ({ fontColour, backgroundColour, grow }) => ({
    color: fontColour,
    backgroundColor: backgroundColour,
    flexGrow: grow ? 1 : 0
  })
);
