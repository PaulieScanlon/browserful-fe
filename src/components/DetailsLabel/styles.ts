import styled from 'react-emotion';
import { scaffolding, colours, transitionBuilder } from '../../theme';
import { font } from '../../typography';

interface IProps {
  width?: number | string;
}

export const LabelWrapper = styled.div({
  label: 'label-wrapper',
  display: 'flex',
  alignItems: 'center',
  minHeight: '56px',
  width: '100%'
});

export const LabelText = styled.span({
  label: 'label-text',
  fontFamily: font.fontFamily,
  fontSize: '16px',
  lineHeight: '18px',
  marginRight: scaffolding.gutterLg
});

export const BoldText = styled.span({
  label: 'bold-text',
  fontWeight: 'bold',
  marginRight: scaffolding.gutterLg
});

export const BarChartWrapper = styled.div({
  label: 'bar-chart-wrapper',
  position: 'relative',
  height: '6px',
  width: '10%',
  maxWidth: '100px',
  backgroundColor: colours.offWhite
});

export const BarChart = styled.div<IProps>(
  {
    label: 'bar-chart',
    position: 'absolute',
    height: '100%',
    backgroundColor: colours.green,
    transition: transitionBuilder('width')
  },
  ({ width }) => ({
    width: `${width}%`
  })
);
