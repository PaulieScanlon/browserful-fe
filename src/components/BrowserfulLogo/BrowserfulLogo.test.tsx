import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';
expect.extend(createMatchers(emotion));

import { colours } from '../../theme';

import { BrowserfulLogo } from './';

import { LogoWrapper, SvgWrapper } from './styles';
import { H5 } from '../../typography';

const shallowDefault = props => {
  return shallow(<BrowserfulLogo {...props} />);
};

const mountDefault = props => {
  return mount(<BrowserfulLogo {...props} />);
};

describe('<BrowserfulLogo/>', () => {
  it('renders correctly', () => {
    const wrapper = shallowDefault({});
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('has a <LogoWrapper/>', () => {
    const wrapper = shallowDefault({});
    const logoWrapper = wrapper.find(LogoWrapper);
    expect(logoWrapper).toHaveLength(1);
  });
  it('has an <SvgWrapper/>', () => {
    const wrapper = shallowDefault({});
    const svgWrapper = wrapper.find(SvgWrapper);
    expect(svgWrapper).toHaveLength(1);
  });
  it('renders text if showText:true', () => {
    const wrapper = mountDefault({ showText: true });
    const h5 = wrapper.find(H5);
    expect(h5).toHaveLength(1);
  });
  it('renders logo with correct colour with logoColour', () => {
    const wrapper = mountDefault({ logoColour: colours.teal });
    const svg = wrapper.find('svg');
    expect(wrapper.props().logoColour).toEqual(colours.teal);
    expect(svg.props().style.fill).toEqual(colours.teal);
  });
  it('renders typography with correct colour with fontColour', () => {
    const wrapper = mountDefault({
      showText: true,
      fontColour: colours.greyLight
    });
    const h5 = wrapper.find(H5);
    expect(wrapper.props().fontColour).toEqual(colours.greyLight);
    expect(h5.props().fontColour).toEqual(colours.greyLight);
    expect(h5).toHaveStyleRule('color', `${colours.greyLight}`);
  });
});
