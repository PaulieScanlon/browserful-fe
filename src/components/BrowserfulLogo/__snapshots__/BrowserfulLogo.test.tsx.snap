// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrowserfulLogo/> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BrowserfulLogo
    fontColour="#404040"
    logoColour="#ff5ca3"
    showText={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <SvgWrapper>
          <svg
            height="23.5"
            style={
              Object {
                "fill": "#ff5ca3",
              }
            }
            width="23.5"
            x="0px"
            y="0px"
          >
            <circle
              cx="9.6"
              cy="9.6"
              r="9.6"
            />
            <g>
              <circle
                cx="20.3"
                cy="20.3"
                r="3.2"
              />
            </g>
          </svg>
        </SvgWrapper>,
        <BrowserfulLogoText
          display="inline-flex"
          fontColour="#404040"
        >
          Browserful
        </BrowserfulLogoText>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <svg
            height="23.5"
            style={
              Object {
                "fill": "#ff5ca3",
              }
            }
            width="23.5"
            x="0px"
            y="0px"
          >
            <circle
              cx="9.6"
              cy="9.6"
              r="9.6"
            />
            <g>
              <circle
                cx="20.3"
                cy="20.3"
                r="3.2"
              />
            </g>
          </svg>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <circle
                cx="9.6"
                cy="9.6"
                r="9.6"
              />,
              <g>
                <circle
                  cx="20.3"
                  cy="20.3"
                  r="3.2"
                />
              </g>,
            ],
            "height": "23.5",
            "style": Object {
              "fill": "#ff5ca3",
            },
            "width": "23.5",
            "x": "0px",
            "y": "0px",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "cx": "9.6",
                "cy": "9.6",
                "r": "9.6",
              },
              "ref": null,
              "rendered": null,
              "type": "circle",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <circle
                  cx="20.3"
                  cy="20.3"
                  r="3.2"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "cx": "20.3",
                  "cy": "20.3",
                  "r": "3.2",
                },
                "ref": null,
                "rendered": null,
                "type": "circle",
              },
              "type": "g",
            },
          ],
          "type": "svg",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Browserful",
          "display": "inline-flex",
          "fontColour": "#404040",
        },
        "ref": null,
        "rendered": "Browserful",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <SvgWrapper>
            <svg
              height="23.5"
              style={
                Object {
                  "fill": "#ff5ca3",
                }
              }
              width="23.5"
              x="0px"
              y="0px"
            >
              <circle
                cx="9.6"
                cy="9.6"
                r="9.6"
              />
              <g>
                <circle
                  cx="20.3"
                  cy="20.3"
                  r="3.2"
                />
              </g>
            </svg>
          </SvgWrapper>,
          <BrowserfulLogoText
            display="inline-flex"
            fontColour="#404040"
          >
            Browserful
          </BrowserfulLogoText>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <svg
              height="23.5"
              style={
                Object {
                  "fill": "#ff5ca3",
                }
              }
              width="23.5"
              x="0px"
              y="0px"
            >
              <circle
                cx="9.6"
                cy="9.6"
                r="9.6"
              />
              <g>
                <circle
                  cx="20.3"
                  cy="20.3"
                  r="3.2"
                />
              </g>
            </svg>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <circle
                  cx="9.6"
                  cy="9.6"
                  r="9.6"
                />,
                <g>
                  <circle
                    cx="20.3"
                    cy="20.3"
                    r="3.2"
                  />
                </g>,
              ],
              "height": "23.5",
              "style": Object {
                "fill": "#ff5ca3",
              },
              "width": "23.5",
              "x": "0px",
              "y": "0px",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "cx": "9.6",
                  "cy": "9.6",
                  "r": "9.6",
                },
                "ref": null,
                "rendered": null,
                "type": "circle",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <circle
                    cx="20.3"
                    cy="20.3"
                    r="3.2"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "cx": "20.3",
                    "cy": "20.3",
                    "r": "3.2",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "circle",
                },
                "type": "g",
              },
            ],
            "type": "svg",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Browserful",
            "display": "inline-flex",
            "fontColour": "#404040",
          },
          "ref": null,
          "rendered": "Browserful",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
